sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -505,3 +506,150 @@ case class ArrayMax(child: Expression) extends UnaryExpression with ImplicitCast
 
   override def prettyName: String = "array_max"
 }
+
+abstract class ArraySetUtils extends BinaryExpression with ExpectsInputTypes {
+  val kindUnion = 1
```

Comment:
This should be in `ArraySetUtils` object?

Reply:
sure